backend/src/services/perplexity_search_service.py

```
@@ -570,4 +584,4 @@ async def _make_perplexity_request(self, prompt: str) -> str:
                     return data["choices"][0]["message"]["content"]
                 else:
                     error_text = await response.text()
-                    raise Exception(f"Perplexity API error: {response.status} - {error_text}")
\ No newline at end of file
+                    raise Exception(f"Perplexity API error: {response.status} - {error_text}")
```

Ruff (0.13.1)

Create your own exception

(TRY002)